neo/SmartContract/InteropService.NEO.cs

```diff
@@ -60,7 +63,82 @@ private bool Native_Deploy(ApplicationEngine engine)
             return true;
         }
 
-        private bool Header_GetVersion(ApplicationEngine engine)
+        private static bool Crypto_CheckSig(ApplicationEngine engine)
+        {
+            byte[] pubkey = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            byte[] signature = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+
+            try
+            {
+                engine.CurrentContext.EvaluationStack.Push(Crypto.Default.VerifySignature(engine.ScriptContainer.GetHashData(), signature, pubkey));
+            }
+            catch (ArgumentException)
+            {
+                engine.CurrentContext.EvaluationStack.Push(false);
+            }
+            return true;
+        }
+
+        private static bool Crypto_CheckMultiSig(ApplicationEngine engine)
+        {
+            int n;
+            byte[][] pubkeys;
+            StackItem item = engine.CurrentContext.EvaluationStack.Pop();
+
+            if (item is VMArray array1)
+            {
+                pubkeys = array1.Select(p => p.GetByteArray()).ToArray();
+                n = pubkeys.Length;
+                if (n == 0) return false;
+            }
+            else
+            {
+                n = (int)item.GetBigInteger();
+                if (n < 1 || n > engine.CurrentContext.EvaluationStack.Count) return false;
+                pubkeys = new byte[n][];
+                for (int i = 0; i < n; i++)
+                    pubkeys[i] = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            }
+
+            int m;
+            byte[][] signatures;
+            item = engine.CurrentContext.EvaluationStack.Pop();
+            if (item is VMArray array2)
+            {
+                signatures = array2.Select(p => p.GetByteArray()).ToArray();
+                m = signatures.Length;
+                if (m == 0 || m > n) return false;
+            }
+            else
+            {
+                m = (int)item.GetBigInteger();
+                if (m < 1 || m > n || m > engine.CurrentContext.EvaluationStack.Count) return false;
+                signatures = new byte[m][];
+                for (int i = 0; i < m; i++)
+                    signatures[i] = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            }
+            byte[] message = engine.ScriptContainer.GetHashData();
+            bool fSuccess = true;
+            try
+            {
+                for (int i = 0, j = 0; fSuccess && i < m && j < n;)
+                {
+                    if (Crypto.Default.VerifySignature(message, signatures[i], pubkeys[j]))
+                        i++;
+                    j++;
+                    if (m - i > n - j)
+                        fSuccess = false;
+                }
+            }
+            catch (ArgumentException)
```

Comment:
In order to prevent future problems if we throw other exceptions, i prefer to return false for all of them
```suggestion
            catch
```

Reply:
`ArgumentException` means that the validation failed, so push `false`. Other exceptions should be thrown directly.

Reply:
But i am scared about what happens if with an update, dotnet start to throw other kind of exceptions in the future, maybe is possible to have different behaviors in the future.

Reply:
This is almost impossible. If it happens, we can also upgrade.